ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactChoreographer.java

```diff
@@ -46,6 +46,11 @@
      * Events that make JS do things.
      */
     TIMERS_EVENTS(3),
+
+    /**
+     * Event used to dispatch the idle callback. Called after all other work is done for the frame.
```

Comment:
I would rephrase this to say that these are called when there is no other UI work that will be dispatched to JS this frame: after this is called, there is still the layout/draw pass of the frame on the UI thread.